architecture/Metatransactions.md

```diff
@@ -0,0 +1,68 @@
+# Metatransactions in the CDapp
+
+Metatransactions (MetaTX) present a paradigm shift in the blockchain world, especially in enhancing user experience by removing the need for users to pay gas fees. The CDapp leverages this concept by introducing a two-pronged approach involving the broadcaster service and specially designed contract methods.
+
+## What are Metatransactions?
+
+Metatransactions encapsulate the original transaction and relay it via a third-party entity which takes responsibility for the gas fee. This enables seamless interactions, particularly beneficial for new users unfamiliar with the concept of gas or those unwilling to spend on every transaction.
+
+### Benefits
+
+User Friendliness: Users without ETH can still perform transactions.
+Flexibility: Allows dApps like CDapp to decide when and how to bear the gas costs, possibly leading to promotional or free transaction periods.
+Adoption: Removes a major hurdle for new users by simplifying interactions with blockchain-based applications.
+
+## Broadcaster Service Middleware
+
+The Broadcaster Service is not just an intermediary, but a critical component in the Metatransaction architecture.
+
+### Workflow
+- The user initiates a transaction on the CDapp.
+- Instead of sending the transaction directly to the blockchain, it's encoded and packaged.
```

Comment:
Could you add a footnote here that goes deeper into how it's `encoded and packaged`?

Would be good to have a bit more detail or links if people want to delve further past the high level hand waving :slightly_smiling_face:

Reply:
Or at least a link to the file where this actually happens